awswrangler/timestream.py

```diff
@@ -85,6 +91,33 @@ def _write_batch(
     return []
 
 
+@engine.dispatch_on_engine
+def _write_df(
+    df: pd.DataFrame,
+    executor: Any,
+    database: str,
+    table: str,
+    cols_names: List[str],
+    measure_cols_names: List[str],
+    measure_types: List[str],
+    version: int,
+    boto3_session: Optional[boto3.Session] = None,
+) -> List[Dict[str, str]]:
+    batches: List[List[Any]] = _utils.chunkify(lst=_df2list(df=df), max_length=100)
```

Comment:
Here the split modin dataframe block reference id is received. I assume modin/ray is smart enough to avoid a shuffle (i.e. pulling a block from one worker to another) and would instead run the remote functions (`_write_df` and `_write_batch`) in the worker where the block already exists...

Reply:
The blocks would be broken down into batches and sent to workers so unfortunately some shuffle or rather copy will inevitably happen. One thing I'm afraid of is `max_length=100` - these would be too fine-grained tasks, might not be worth it because of the overhead

Reply:
Fair point, the load test on 64,000 rows was fine but let me check with an even larger one tomorrow